test/core/utils/qsa.js

```diff
@@ -48,184 +48,229 @@ function getTestDom() {
-    var result = axe.utils.querySelectorAllFilter(
-      dom,
-      'div:not([data-a11yhero=wilco])'
-    );
-    assert.equal(result.length, 3);
-  });
-  it('should find nodes using :not selector with bogus id', function() {
-    var result = axe.utils.querySelectorAllFilter(dom, 'div:not(#thangy)');
-    assert.equal(result.length, 3);
-  });
-  it('should find nodes using :not selector with attribute', function() {
-    var result = axe.utils.querySelectorAllFilter(dom, 'div:not([id])');
-    assert.equal(result.length, 2);
-  });
-  it('should find nodes hierarchically using :not selector', function() {
-    var result = axe.utils.querySelectorAllFilter(dom, 'div:not(.first) li');
-    assert.equal(result.length, 2);
-  });
-  it('should find same nodes hierarchically using more :not selector', function() {
-    var result = axe.utils.querySelectorAllFilter(
-      dom,
-      'div:not(.first) li:not(.breaking)'
-    );
-    assert.equal(result.length, 2);
-  });
-  it('should NOT find nodes hierarchically using :not selector', function() {
-    var result = axe.utils.querySelectorAllFilter(
-      dom,
-      'div:not(.second) li:not(.breaking)'
-    );
-    assert.equal(result.length, 0);
-  });
-  it('should find nodes using ^= attribute selector', function() {
-    var result = axe.utils.querySelectorAllFilter(dom, '[class^="sec"]');
-    assert.equal(result.length, 1);
-  });
-  it('should find nodes using $= attribute selector', function() {
-    var result = axe.utils.querySelectorAllFilter(dom, '[id$="ne"]');
-    assert.equal(result.length, 3);
-  });
-  it('should find nodes using *= attribute selector', function() {
-    var result = axe.utils.querySelectorAllFilter(dom, '[role*="t"]');
-    assert.equal(result.length, 2);
-  });
-  it('should put it all together', function() {
-    var result = axe.utils.querySelectorAllFilter(
-      dom,
-      '.first[data-a11yhero="faulkner"] > ul li.breaking'
-    );
-    assert.equal(result.length, 2);
-  });
-  it('should find an element only once', function() {
-    var divs = axe.utils.querySelectorAllFilter(dom, 'div');
-    var ones = axe.utils.querySelectorAllFilter(dom, '#one');
-    var divOnes = axe.utils.querySelectorAllFilter(dom, 'div, #one');
-
-    assert.isBelow(
-      divOnes.length,
-      divs.length + ones.length,
-      'Elements matching both parts of a selector should not be included twice'
-    );
-  });
-  it('should return nodes sorted by document position', function() {
-    var result = axe.utils.querySelectorAllFilter(dom, 'ul, #one');
-    assert.equal(result[0].actualNode.nodeName, 'UL');
-    assert.equal(result[1].actualNode.nodeName, 'DIV');
-    assert.equal(result[2].actualNode.nodeName, 'UL');
-  });
-  it('should filter the returned nodes when passed a filter function', function() {
-    var result = axe.utils.querySelectorAllFilter(dom, 'ul, #one', function(
-      node
-    ) {
-      return node.actualNode.nodeName !== 'UL';
+
```

Comment:
This file is mostly just adding the wrapping describe. The first 30 lines are new code, then one additional if statement added in the middle (marked in another comment). The rest is no change